opaque_keys/edx/locator.py

```diff
@@ -1330,3 +1330,67 @@ def to_deprecated_list_repr(self):
 
 # Register AssetLocator as the deprecated fallback for AssetKey
 AssetKey.set_deprecated_fallback(AssetLocator)
+
+
+class AggregateCourseLocator(AggregateCourseKey):    # pylint: disable=abstract-method
```

Comment:
a) Sorry I forgot to mention that the complete format of course key has version, e.g. `course-v2:edX+DemoX`
b) I have added the method `make_course_key_v2` in `CourseKey` which will return `CourseKeyV2` locator. Also added the method `make_course_run_key` in `CourseKeyV2` which will return a `CourseKey` locator `CourseLocator` from the provided course run.
@cpennington thanks for the suggestions. Please have a look on the changes and let me know if I have missed anything.